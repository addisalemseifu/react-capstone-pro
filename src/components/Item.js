import React from 'react';
import { Link } from 'react-router-dom';
import { ArrowCircleRight } from './icons';

export default function Item({
  // eslint-disable-next-line react/prop-types
  date, sunrise, sunset, id,
}) {
  return (
    <div className="item-container">

      <Link
        to={`/details/${id}`}
      >
        <ArrowCircleRight />
      </Link>
      <div className="date">
        <h3>Forcast Date</h3>
        <h4>{date}</h4>
      </div>
      <div className="day-time">
        <h5>
          Sunrise:-
          {sunrise}
        </h5>
        <h5>
          Sunset:-
          {sunset}
        </h5>

      </div>
    </div>
  );
}
