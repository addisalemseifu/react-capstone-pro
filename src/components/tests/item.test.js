import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Item from '../Item';
import '@testing-library/jest-dom';

describe('item', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Item /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
