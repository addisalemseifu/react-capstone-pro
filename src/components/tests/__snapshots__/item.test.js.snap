// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item should matche DOM Snapshot 1`] = `
<div
  className="item-container"
>
  <a
    href="/details/undefined"
    onClick={[Function]}
  >
    <svg
      className="w-6 h-6"
      fill="none"
      stroke="currentColor"
      strokeWidth="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </a>
  <div
    className="date"
  >
    <h3>
      Forcast Date
    </h3>
    <h4 />
  </div>
  <div
    className="day-time"
  >
    <h5>
      Sunrise:-
    </h5>
    <h5>
      Sunset:-
    </h5>
  </div>
</div>
`;
